nixos/modules/services/web-servers/nginx/default.nix

```
@@ -745,20 +748,27 @@ in
       }
 
       {
-        assertion = all (conf: with conf;
-          !(addSSL && (onlySSL || enableSSL)) &&
-          !(forceSSL && (onlySSL || enableSSL)) &&
-          !(addSSL && forceSSL)
+        assertion = all (host: with host;
+          count id [ addSSL (onlySSL || enableSSL) forceSSL rejectSSL ] <= 1
         ) (attrValues virtualHosts);
```

```suggestion
        assertion = all (host: with host; count id [ addSSL (onlySSL || enableSSL) forceSSL rejectSSL ] <= 1) (attrValues virtualHosts);
```

You see the problem?

---

No.

---

I didn't put the entire assert together on the first try because it is to long and complex to overlook at first sight but that could also be the github diff GUI.

---

Why do you want it on a single line? I find that much harder to read.